src/uu/head/src/head.rs

```diff
@@ -184,7 +186,9 @@ fn arg_iterate<'a>(
                 None => Ok(Box::new(vec![first, second].into_iter().chain(args))),
             }
         } else {
-            Err(HeadError::BadEncoding)
+            // The second argument contains non-UTF-8 sequences, so it can't be an obsolete option
+            // like "-5". Treat it as a regular file argument.
+            Ok(Box::new(vec![first, second].into_iter().chain(args)))
```

Comment:
You can avoid code duplication (please restore the comment though):

```suggestion
        if let Some(s) = second.to_str() &&
            let Some(v) = parse::parse_obsolete(s) {
            match v {
                Ok(iter) => Ok(Box::new(vec![first].into_iter().chain(iter).chain(args))),
                Err(parse::ParseError) => Err(HeadError::ParseError(
                    translate!("head-error-bad-argument-format", "arg" => s.quote()),
                )),
            }
        } else {
            // comment here
            Ok(Box::new(vec![first, second].into_iter().chain(args)))
        }
```